src/backend/generic_backend.cc

```diff
@@ -0,0 +1,48 @@
+#include "backend/generic_backend.h"
+
+#include "ranges.h"
+
+namespace celerity::detail::backend_detail {
+
+void memcpy_strided_device_generic(sycl::queue& queue, const void* source_base_ptr, void* target_base_ptr, size_t elem_size, const sycl::range<1>& source_range,
+    const sycl::id<1>& source_offset, const sycl::range<1>& target_range, const sycl::id<1>& target_offset, const sycl::range<1>& copy_range) {
+	const size_t line_size = elem_size * copy_range[0];
+	auto evt = queue.memcpy(static_cast<char*>(target_base_ptr) + elem_size * get_linear_index(target_range, target_offset),
+	    static_cast<const char*>(source_base_ptr) + elem_size * get_linear_index(source_range, source_offset), line_size);
+	evt.wait();
+}
+
+// TODO Optimize for contiguous copies?
+void memcpy_strided_device_generic(sycl::queue& queue, const void* source_base_ptr, void* target_base_ptr, size_t elem_size, const sycl::range<2>& source_range,
+    const sycl::id<2>& source_offset, const sycl::range<2>& target_range, const sycl::id<2>& target_offset, const sycl::range<2>& copy_range) {
+	const auto source_base_offset = get_linear_index(source_range, source_offset);
+	const auto target_base_offset = get_linear_index(target_range, target_offset);
+	const size_t line_size = elem_size * copy_range[1];
+	std::vector<sycl::event> wait_list;
```

Comment:
Consider explicitly giving the vector size as `copy_range[0]` here.

Reply:
Okay, sure.